pkg/auth/oidc/impl.go

```diff
@@ -91,6 +95,32 @@ type usernameSetter interface {
 	SetUsername(username string)
 }
 
+// checkOrigin checks if the "Origin" header is present and corresponds to the
+// request host, and returns nil if so, otherwise an error is returned and an
+// error response is written to w.
+func checkOrigin(w http.ResponseWriter, r *http.Request) error {
+	origin := r.Header.Get("Origin")
+	if origin == "" {
+		http.Error(w, "Bad request", http.StatusBadRequest)
+		return errors.New("no Origin header")
+	}
```

Comment:
The origin header is only set by "user-agents (browsers)" so with this logic you woul break api consumers.  This test should pass if there is no origin or referer set (return nil).

Reply:
API consumers are using an X.509 certificate to authenticate, so they never reach this code path.